src/AppBundle/Tests/Controller/TodoControllerTest.php

```diff
@@ -0,0 +1,246 @@
+<?php
+
+namespace AppBundle\Tests\Controller;
+
+use Symfony\Bundle\FrameworkBundle\Client;
+use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
+
+class TodoControllerTest extends WebTestCase
+{
+    /**
+     * Login with FOSUserBundle.
+     * @param $client
+     * @param $username
+     * @param $password
+     */
+    public function doLogin(Client $client, $username = "testaccount", $password = "testaccount") {
+        $crawler = $client->request('GET', '/login');
+        $form = $crawler->selectButton('_submit')->form(array(
+            '_username'  => $username,
+            '_password'  => $password,
+        ));
+        $client->submit($form);
+
+        //echo $client->getResponse();
+
+        $this->assertTrue($client->getResponse()->isRedirect());
+
+        $crawler = $client->followRedirect();
+
+        //echo $client->getResponse();
+    }
+
+    /**
+     * Get the response body as a JSON object.
+     *
+     * If it was unable to parse the body (malformed JSON or JSON array, or no body) it returns a string.
+     *
+     * @param Client $client
+     * @return array|string
+     */
+    public function responseAsJSON($client)
+    {
+        $result = '';
+        $content = $client->getResponse()->getContent();
+        try {
+            if (!empty($content)) {
+                $result = json_decode($content, true);
+            }
+        } catch (\Exception $e) {
+            $result = $content;
+        }
+        return $result;
+    }
+
+    public function testTodoSimple()
```

Comment:
This is a very big test case, being a functional test it's fine, but it needs a doc comment listing what it's testing.